### DISCUSSION TITLE
Keep content page within window bounds

### DISCUSSION THREAD

**Discussion by LtKlaus** [lepoco/wpfui#822] [Q&A] [answered]
This is my first time using WPF UI and I really like how it looks and how easy it's been to implement so far. 

Because I'm new to this, I'm not sure the best way to explain my issue. I have a page that shows a DataGrid with some buttons above it for adding new items, editing the selected item, etc. When the DataGrid has a lot of rows, the DataGrid stretches off the bottom of the window and the page shows a scrollbar. Using the scrollbar scrolls the full page and the buttons above the grid go off the top of the window. 

The root control in the page is a grid with two rows. The height of the first row is Auto and the second is *. What I want is the DataGrid to  just fill the screen and show a scrollbar inside the DataGrid instead of the full page. 

I can set a height on my root grid of the DataGrid and the DataGrid will show it's scrollbar. That just makes it a fixed height though and it no longer resizes with my application.

Hope that makes sense.

**Comment by LtKlaus** [comment]
Looks like I got it figured out. I changed the Page to a UserControl and it sizes the way I want now

**Reply by gsnipedev** [reply]
I dont think using UserControl instead of Page for a Page is a good practice, im stuck in this problem also.

**Comment by Mithicor** [comment, marked as answer]
Instead of changing the Page to a UserControl, have you tried adding the Page Property `ScrollViewer.CanContentScroll="False"`?

**Reply by LtKlaus** [reply]
Finally got back to working with this again and I can confirm this does work. Thanks.
